File: util/src/main/java/org/camposmdev/util/Log.java
package org.camposmdev.util;

import com.almasb.fxgl.logging.Logger;

public class Log {
    private Log() {}

    private static Class<?> getCallerClass() {
        StackTraceElement[] stackTrace = Thread.currentThread().getStackTrace();
        try {
            return Class.forName(stackTrace[3].getClassName());
        } catch (ClassNotFoundException e) {
            return Object.class;
        }
    }

    public static void debug(String message) {
        Logger.get(getCallerClass()).debug(message);
    }

    public static void debugf(String format, Object... args) {
        Logger.get(getCallerClass()).debugf(format, args);
    }

    public static void info(String message) {
        Logger.get(getCallerClass()).info(message);
    }

    public static void infof(String format, Object... args) {
        Logger.get(getCallerClass()).infof(format, args);
    }

    public static void warn(String message) {
        Logger.get(getCallerClass()).warning(message);
    }

    public static void warn(String message, Throwable error) {
        Logger.get(getCallerClass()).fatal(message, error);
    }

    public static void warnf(String format, Object... args) {
        Logger.get(getCallerClass()).warningf(format, args);
    }

    public static void fatal(String message) {
        Logger.get(getCallerClass()).fatal(message);
    }

    public static void fatal(String message, Throwable error) {
        Logger.get(getCallerClass()).fatal(message, error);
    }

    public static void fatalf(String format, Object... args) {
        Logger.get(getCallerClass()).fatalf(format, args);
    }
}

File: client/src/main/java/org/camposmdev/client/model/Log.kt
package org.camposmdev.client.model

import com.almasb.fxgl.logging.Logger;
import org.camposmdev.client.ui.FXUtil

class Log private constructor() {
    private companion object {
        @JvmStatic
        private val callerClass: Class<*>
            get() {
                val stackTrace = Thread.currentThread().stackTrace
                return try {
                    Class.forName(stackTrace[4].className)
                } catch (e: ClassNotFoundException) {
                    FXUtil::class.java
                }
            }

        @JvmStatic
        fun debug(message: String) {
            Logger.get(callerClass).debug(message)
        }

        @JvmStatic
        fun debugf(format: String, vararg args: Any) {
            Logger.get(callerClass).debugf(format, *args)
        }

        @JvmStatic
        fun info(message: String) {
            Logger.get(callerClass).info(message)
        }

        @JvmStatic
        fun infof(format: String, vararg args: Any) {
            Logger.get(callerClass).infof(format, *args)
        }

        @JvmStatic
        fun warn(message: String) {
            Logger.get(callerClass).warning(message)
        }

        @JvmStatic
        fun warn(message: String, error: Throwable) {
            Logger.get(callerClass).fatal(message, error)
        }

        @JvmStatic
        fun warnf(format: String, vararg args: Any) {
            Logger.get(callerClass).warningf(format, *args)
        }

        @JvmStatic
        fun fatal(message: String) {
            Logger.get(callerClass).fatal(message)
        }

        @JvmStatic
        fun fatal(message: String, error: Throwable) {
            Logger.get(callerClass).fatal(message, error)
        }

        @JvmStatic
        fun fatalf(format: String, vararg args: Any) {
            Logger.get(callerClass).fatalf(format, *args)
        }
    }
}
